src/overlays/actors/ovl_Obj_Lightswitch/z_obj_lightswitch.c

```diff
@@ -27,47 +48,435 @@ const ActorInit Obj_Lightswitch_InitVars = {
+        this->toggleDelay <= 0) {
+        this->timer--;
+        Math_ApproxS(&this->flameRingRotSpeed, 0, 0xA);
+        this->flameRingRot += this->flameRingRotSpeed;
+        this->red = this->timer * (((255 - 155) << 6) / 20) + (155 << 6);
+        this->green = this->timer * (((255 - 125) << 6) / 20) + (125 << 6);
+        if (this->timer <= 0) {
+            ObjLightswitch_OffInit(this);
+        } else if (this->timer == 15) {
+            this->faceTextureIndex = EYES_CLOSED;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_FOOT_SWITCH);
+        }
+    }
+}
+
+void ObjLightswitch_DisappearDelayInit(ObjLightswitch* this) {
+    this->actionFunc = ObjLightswitch_DisappearDelay;
+    this->toggleDelay = 100;
+}
+
+void ObjLightswitch_DisappearDelay(ObjLightswitch* this, GlobalContext* globalCtx) {
+    if (func_8005B198() == this->actor.type || this->toggleDelay <= 0) {
+        ObjLightswitch_DisappearInit(this);
+    }
+}
+
+void ObjLightswitch_DisappearInit(ObjLightswitch* this) {
+    this->actionFunc = ObjLightswitch_Disappear;
+    this->alpha = 255 << 6;
+}
+
+void ObjLightswitch_Disappear(ObjLightswitch* this, GlobalContext* globalCtx) {
+    this->alpha -= 200;
+    ObjLightswitch_SpawnDisappearEffects(this, globalCtx);
+    if (this->alpha < 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void ObjLightswitch_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjLightswitch* this = THIS;
+    Actor* child;
+
+    if (globalCtx) {}
+
+    if (this->toggleDelay > 0) {
+        this->toggleDelay--;
+    }
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->actor.update != NULL) {
+        if ((this->actor.params & 1) == 1) {
+            child = this->actor.child;
+            this->actor.posRot.pos.x = child->posRot.pos.x;
+            this->actor.posRot.pos.y = child->posRot.pos.y + 60.0f;
+            this->actor.posRot.pos.z = child->posRot.pos.z;
+            Actor_SetHeight(&this->actor, 0.0f);
+        }
+        this->prevFrameACflags = this->col.base.acFlags;
+        this->col.base.acFlags &= ~2;
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->col.base);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->col.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lightswitch/func_80B96D00.s")
+void ObjLightswitch_DrawOpa(ObjLightswitch* this, GlobalContext* globalCtx) {
+    Actor* child;
+    Vec3f pos;
+    Vec3s rot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lightswitch/func_80B96D80.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_obj_lightswitch.c", 809);
+    func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Lightswitch/ObjLightswitch_Init.s")
+    /*
+    gDPSetEnvColor(oGfxCtx->polyOpa.p++, (u8)(this->red >> 6), (u8)(this->green >> 6), (u8)(this->blue >> 6),
```

Comment:
This comment can be removed unless there's some significance to it I don't know

Reply:
it's what the line below should look like but I had to do the bitshifting manually to get it to match, should I still remove it? or make it clearer why I left it